### PR TITLE
Collect ODF Provider and Client resources

### PR DESCRIPTION
**Pull request by Mrudraia**
This PR will add additional missing resources in ODF provider cluster and consumer cluster. In provider cluster the following resources YAMLs, OC_OUPUTs, OC_DESC are collected.

In Provider Cluster

- storage consumers
- storage class requests
- storage profiles

 In Consumer cluster

- storageclients
- storageclassclaims
- ocs-client-operator logs, pods, api-server logs etc
- ceph command outputs

### CODE REVIEW

File: collection-scripts/gather [hunk 1]
```diff
@@ -22,11 +22,14 @@ pre-install.sh ${BASE_COLLECTION_PATH}
 gather_namespaced_resources ${BASE_COLLECTION_PATH}
 gather_clusterscoped_resources ${BASE_COLLECTION_PATH}
 gather_noobaa_resources ${BASE_COLLECTION_PATH}
+gather_odf_client ${BASE_COLLECTION_PATH}
```
Comment (black-dragon74) [inline collection-scripts/gather:25]: `gather_odf_client` first called here.

Comment (Mrudraia) [reply collection-scripts/gather:25]: With condition, we are unable to get client CRs, pods, yamls when both provider and client are deployed in a single cluster. 
For cluster with only provider installed, the gather_odf_client will fail and only provider CRs are fetched

**Review by yati1998 — APPROVED** [review]
LGTM

**Comment by leelavg** [comment]
@Mrudraia pls rebase properly, I'm seeing unrelated commits, thanks.

**Comment by black-dragon74** [comment]
> @Mrudraia pls rebase properly, I'm seeing unrelated commits, thanks.

Holding just for that.

File: collection-scripts/gather_odf_client [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+BASE_COLLECTION_PATH=${BASE_COLLECTION_PATH:-"$(pwd)"}
+
+
+STORAGE_CLIENT_NAMESPACE=$(oc get storageclients -A --no-headers | awk '{print $1}')
+CLIENT_OPERATOR_NAMESPACE=$(oc get subscriptions -A --no-headers | grep ocs-client-operator | awk '{print $1}')
+
+# collection path for OC commands
+mkdir -p "${BASE_COLLECTION_PATH}/odf-client/oc_output/"
+
+# Command List
```
Comment (leelavg) [inline collection-scripts/gather_odf_client:14]: irrespective of provider (not/)collecting, add storageclass to all these lists.

File: collection-scripts/gather_odf_client [hunk 3]
```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env bash
```
Comment (leelavg) [inline collection-scripts/gather_odf_client:1]: I believe you are missing storageclassclaims

File: collection-scripts/gather_odf_client [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+commands_desc+=("pods")
+commands_desc+=("deployments")
+commands_desc+=("secrets")
+commands_desc+=("configmaps")
+commands_desc+=("cronjobs")
+commands_desc+=("subscription")
+commands_desc+=("storageclients")
+
+
+for INSTALL_NAMESPACE in  $STORAGE_CLIENT_NAMESPACE $CLIENT_OPERATOR_NAMESPACE ; do
```
Comment (leelavg) [inline collection-scripts/gather_odf_client:45]: we only need storageclient CRs from storage_client_namespace

File: collection-scripts/gather_odf_client [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+     done
+
+     # Run the Collection of OC desc commands
+     for command_desc in "${commands_desc[@]}"; do
+          dbglog "collecting oc describe command ${command_desc}"
+          COMMAND_OUTPUT_FILE=${BASE_COLLECTION_PATH}/namespaces/${INSTALL_NAMESPACE}/oc_output/${command_desc// /_}
+          # shellcheck disable=SC2086
+          { oc describe ${command_desc} -n ${INSTALL_NAMESPACE}; } >>"${COMMAND_OUTPUT_FILE}" 
+     done
+done
```
Comment (leelavg) [inline collection-scripts/gather_odf_client:72]: add new line

**Review by leelavg — CHANGES_REQUESTED** [review]

File: collection-scripts/gather_odf_client (hunk 5 above)
Comment (Mrudraia) [reply collection-scripts/gather_odf_client:74]: Added new line

File: collection-scripts/gather_odf_client (hunk 4 above)
Comment (Mrudraia) [reply collection-scripts/gather_odf_client:45]: Storage client CRs are captured in must gather

File: collection-scripts/gather_odf_client (hunk 3 above)
Comment (Mrudraia) [reply collection-scripts/gather_odf_client:1]: Storageclassclaims are fetched as a clusterscopedresources. The storageclassclaims created with storageclient are also captured in must-gather

**Comment by black-dragon74** [comment]
@Mrudraia Always sync before you push. If you have to force push, you are doing it wrong. Please rebase properly so the PR includes ONLY your changes.

Regards

**Comment by black-dragon74** [comment]
Please fix shell check errors. You need to use double quotes (`"`) around variables to prevent globbing.

Regards

File: collection-scripts/gather_odf_client (hunk 5 above)
Comment (leelavg) [reply collection-scripts/gather_odf_client:76]: no, I still see the red circle here.

File: collection-scripts/gather_odf_client (hunk 4 above)
Comment (leelavg) [reply collection-scripts/gather_odf_client:45]: - "we only need storageclient CRs" --> we do not need anything else, like pods, deployments, secrets etc from namespaces that has storageclient CRs.
- if you are confused, refer https://github.com/red-hat-storage/odf-must-gather/pull/74#discussion_r1363109932 again

File: collection-scripts/gather_odf_client (hunk 2 above)
Comment (leelavg) [reply collection-scripts/gather_odf_client:14]: any reason resolving this without any response?

File: collection-scripts/gather_odf_client (hunk 3 above)
Comment (leelavg) [reply collection-scripts/gather_odf_client:1]: ack, then I don't see them as part of this PR and I highly doubt they are being collected already.

**Review by leelavg — CHANGES_REQUESTED** [review]

**Comment by leelavg** [comment]
@Mrudraia pls change the base branch to `fusion-hci-4.14`

**Review by leelavg — APPROVED** [review]
/lgtm

**Comment by leelavg** [comment]
@yati1998 can this PR be merged? I'm not sure we need PR checks or not as we are merging to fusion branch.

**Comment by black-dragon74** [comment]
> @yati1998 can this PR be merged? I'm not sure we need PR checks or not as we are merging to fusion branch.

Yes. We do not have any CI pipeline for fusion branch.

**Comment by yati1998** [comment]
Merging the PR, LGTM

**Pull request merged**